competition/models.py

```diff
@@ -101,7 +106,7 @@ def __str__(self):
         return self.name
 
 
-class Event(models.Model):
+class Event(models.Model, permissions.CompetitionPermissionMixin):
```

Comment:
CompetitionPermissionMixin definuje can_user_modify metodu pre competition. Vsetky ostatne modely ktore maju metodu can_user_modify sa len rekurzivne odkazuju na foreign key model az kym sa nedostanu ku Competition modelu ktory obsahuje vsetky groupy ktore mozu model editovat.

Vsetky modely ktore pouzivaju CompetitionPermissionMixin (okrem Competition) aj tak prepisuju metodu can_user_modify a nepouzivaju nic z toho mixinu. Preto tento mixin ani nepotrebuju nie? Ide o modely: Event, Series, Problem, EventRegistration, SemesterRegistration, UnspecifiedPublication, RegistrationLink

Reply:
S tým Mixinom máš asi pravdu dám ho preč. Prečo som ho pôvodne plánoval bolo aby značil, že tá trieda poskytuje tento interface ale asi to je tu zbytočné

Reply:
Myslel som si ze si ten mixin myslel ako interface len narozdiel od inych programovacich jazykov nemozes prinutit ludi predefinovat ho a tak to asi nema velmi zmysel.